// @mikecodeur/react-course-app/dist/components/ProgressBar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ProgressBar;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _LinearProgress = _interopRequireDefault(require("@material-ui/core/LinearProgress"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _Box = _interopRequireDefault(require("@material-ui/core/Box"));

var _constants = require("../utils/constants");

var _LocalStorageHook = _interopRequireDefault(require("./hook/LocalStorageHook"));

var _Fireworks = _interopRequireDefault(require("./qcm/Fireworks"));

function LinearProgressWithLabel(props) {
  // let a:Toto
  //console.log(QCM_RESULT);
  return /*#__PURE__*/_react.default.createElement(_Box.default, {
    display: "flex",
    alignItems: "center"
  }, /*#__PURE__*/_react.default.createElement(_Box.default, {
    width: "100%",
    mr: 1
  }, /*#__PURE__*/_react.default.createElement(_LinearProgress.default, (0, _extends2.default)({
    variant: "determinate"
  }, props))), /*#__PURE__*/_react.default.createElement(_Box.default, {
    minWidth: 35
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    variant: "body2",
    color: "textSecondary"
  }, `${Math.round(props.value)}%`)));
} // const useStyles = makeStyles({
//   root: {
//     width: '40%',
//     paddingLeft:'20px'
//   },
// });


const useStyles = (0, _styles.makeStyles)(theme => (0, _styles.createStyles)({
  root: {
    width: '40%',
    paddingLeft: '20px',
    [theme.breakpoints.down('md')]: {
      display: 'none'
    }
  }
}));

//const QCM_RESULT = 'QcmResult';
function ProgressBar(props) {
  const classes = useStyles();

  const [progress, setProgress] = _react.default.useState(10); //const [projectId] = useLocalStorage<string>(QCM_CURRENT_PROJECT_ID, '');


  const projectId = window.localStorage.getItem(_constants.QCM_CURRENT_PROJECT_ID);
  const [qcmAllResults] = (0, _LocalStorageHook.default)(`${_constants.QCM_RESULT}-${projectId}`, []);

  _react.default.useEffect(() => {
    /*const timer = setInterval(() => {
      setProgress((prevProgress) => (prevProgress >= 100 ? 10 : prevProgress + 10));
    }, 800);
    return () => {
      clearInterval(timer);
    };*/
    //console.log('ProgressBar',qcmAllResults.length/props.exoLength)
    //console.log('ProgressBar',props.exoLength)
    setProgress(qcmAllResults.length / (props.exoLength - 1) * 100);
  }, []);

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement(_Fireworks.default, {
    autoFire: true
  }), /*#__PURE__*/_react.default.createElement(LinearProgressWithLabel, {
    value: progress
  }));
}
